### ISSUE TITLE
False negative ``useless-parent-delegation`` when using the old style super/parent lookup

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [pylint-dev/pylint#5083]
### Bug description

Noticed that with pyupgrade making a new message appear :

```python
class ClassDiadefGenerator(DiaDefGenerator):

    def __init__(self, linker, handler):
        DiaDefGenerator.__init__(self, linker, handler) # no useless-super-delegation
        # super().__init__(linker, handler) # will trigger useless-super-delegation
```

### Pylint output

```shell
No output
```

### Expected behavior

Raise a useless-super-delegation in both case.

### Pylint version

```shell
2.11.2-dev0
```

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for being willing to handle this. I think it's possible to keep only one function and add a condition to handle the case. The diff should not be very big unless I'm mistaken.

**Comment by Sveder** [comment]
Ok, cool, if the diff is small enough I'll just keep it in one function. Thanks!

**Comment by DanielNoord** [comment]
I took a look at this but I am not sure if this would fit `useless-super-delegation`. There is no use of `super` so I think this would be confusing for users if it would emit a warning which mentions it.

**Comment by Pierre-Sassoulas** [comment]
Do you propose to add another message or change the message name ? Maybe, ``useless-function-overidding`` ? We often have to handle both ``super``, ``self``, ``cls``, and the base class to handle all cases. I think generally we keep using the same message maybe because it does not mention ``super`` explicitly. I don't remember exactly which messages right now... maybe ``unused-private-attribute``?

**Comment by Pierre-Sassoulas** [comment]
> It feels like if you're going through the effort of doing this you'll likely know what you're doing?

In my experience it's easy to copy paste code from stackoverflow and don't understand anything about what you're doing. Well case in point, the original example came from pylint and was detected because the code was modified by pyupgrade to use super instead. And I think we('re supposed to) know what we're doing 😄

**Comment by DanielNoord** [comment]
`pyupgrade` doesn't say why it does this in its readme but I think it is because the `mro` of the class equalises `super()` to `DiaDefGenerator` so there is no reason to be explicit.
That would make this a `unnecessary-explicit-parent-reference` message? Either way, it's not a `useless-super-delegation` false negative I think.

**Comment by DanielNoord** [comment]
I understand that, but I don't think that's the "main" issue here. Or at least, we can't emit the super message as there is no super in the original call.

The issue pyupgrade recognises is something we could also warn about. So then I think it's easier to rely on the logic of `uselesss-super` afterwards.

Otherwise we would need:
`useless-parent-delegation` with the same logic and
`useless-explicit-parent` which recognises that `super()` should be used instead.

**Comment by Pierre-Sassoulas** [comment]
How about we rename the message so it's more generic and fix the false negative ? Creating a new check to warn about something that is automatically fixed by another tool is unlikely to attract contributors.

**Comment by DanielNoord** [comment]
> How about we rename the message so it's more generic and fix the false negative ? Creating a new check to warn about something that is automatically fixed by another tool is unlikely to attract contributors.

Yeah, I'd be okay with making it `useless-parent-delegation` and updating the description.

I think `pyupgrade` is less used than `pylint` so I think it still makes sense to add the check. We could even link to `pyupgrade` in its documentation example 😄
Considering we already do the lookup in the `mro` for `useless-parent` it shouldn't be to hard to determine if it is unneeded.

**Comment by Pierre-Sassoulas** [comment]
> I think pyupgrade is less used than pylint so I think it still makes sense to add the check. We could even link to pyupgrade in its documentation example 

Make sense, we have check using isort that is used more than pylint. Let's open another issue though.

**Comment by DanielNoord** [comment]
Opened new issues and renamed this one.

**Comment by Pierre-Sassoulas** [comment]
(The renaming done by Daniel is accurate). We fixed https://github.com/PyCQA/pylint/issues/6953, so now there need to be a message for the following functional test 

```python
class ClassDiadefGenerator(DiaDefGenerator):

    def __init__(self, linker, handler):
        DiaDefGenerator.__init__(self, linker, handler) # [useless-super-delegation]
```

**Comment by Muzammillcoste** [comment]
please assign this issue to me.
